### PR TITLE
Safe default behaviour when no users are specified on trashbin:cleanup

### PR DESCRIPTION
**Pull request by liamdennehy**
* Add option --all-users to explicitly clean all trashbins
* Reject no users on commandline and no --all-users
* Warn when --all-users and userids are specified
#10001 

Signed-off-by: Liam Dennehy <liam@wiemax.net>

### CODE REVIEW

File: apps/files_trashbin/lib/Command/CleanUp.php [hunk 1]
```diff
@@ -77,8 +87,8 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 					$output->writeln("<error>Unknown user $user</error>");
 				}
 			}
-		} else {
-			$output->writeln('Remove all deleted files');
+		} elseif ($input->getOption('all-users')) {
+			$output->writeln('Remove deleted files for all users');
```
Comment (MorrisJobke) [inline apps/files_trashbin/lib/Command/CleanUp.php]: Could you add following `else` block to inform about the needed parameters:

```php
} else {
	throw new InvalidOptionException('Either specify a user_id or --all-users');
}
```

And don't forget to import the exception then as well:

```php
use Symfony\Component\Console\Exception\InvalidOptionException;
```

**Review by MorrisJobke — APPROVED** [review]
Looks good beside my little nitpick 👍

File: apps/files_trashbin/lib/Command/CleanUp.php (hunk 1 above)
Comment (liamdennehy) [reply apps/files_trashbin/lib/Command/CleanUp.php]: Done

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
unit tests need to be adjusted

**Comment by MorrisJobke** [comment]
> unit tests need to be adjusted

Ref:

```
2) OCA\Files_Trashbin\Tests\Command\CleanUpTest::testExecuteAllUsers
Symfony\Component\Console\Exception\InvalidOptionException: Either specify a user_id or --all-users

/drone/src/github.com/nextcloud/server/apps/files_trashbin/lib/Command/CleanUp.php:110
/drone/src/github.com/nextcloud/server/tests/lib/TestCase.php:210
/drone/src/github.com/nextcloud/server/apps/files_trashbin/tests/Command/CleanUpTest.php:206
```

@liamdennehy If you need help, then reach out to us. Would be good to get this in today, as we have feature freeze today.

**Comment by rullzer** [comment]
I took the liberty of fixing the tests so we can merge it :)

**Comment by rullzer** [comment]
Failing tests are unrelated.

**Comment by rullzer** [comment]
Cool stuff @liamdennehy 

Feel free to keep joining IRC.
And also feel free to tackle other issues :rocket: :guitar:

**Comment by liamdennehy** [comment]
Thanks @rullzer, I would have taken ages to figure out how to create those tests!

**Pull request merged**